cloudinit/sources/DataSourceEc2.py

```diff
@@ -731,25 +737,66 @@ def convert_ec2_metadata_network_config(network_md, macs_to_nics=None,
         nic_metadata = macs_metadata.get(mac)
         if not nic_metadata:
             continue  # Not a physical nic represented in metadata
-        nic_cfg = {'type': 'physical', 'name': nic_name, 'subnets': []}
-        nic_cfg['mac_address'] = mac
-        if (nic_name == fallback_nic or nic_metadata.get('public-ipv4s') or
-                nic_metadata.get('local-ipv4s')):
-            nic_cfg['subnets'].append({'type': 'dhcp4'})
-        if nic_metadata.get('ipv6s'):
-            nic_cfg['subnets'].append({'type': 'dhcp6'})
-        netcfg['config'].append(nic_cfg)
+        nic_idx += 1
+        dev_config = {'dhcp4': True, 'dhcp6': False,
+                      'match': {'macaddress': mac.lower()},
+                      'set-name': nic_name}
+        ipv4s = nic_metadata.get('local-ipv4s')
+        ipv6s = nic_metadata.get('ipv6s')
+        if ipv6s:
+            dev_config['dhcp6'] = True
+        # In version < 2018-09-24 local_ipv4s or ipv6s is a str with one IP
+        secondary_ipv4 = bool(isinstance(ipv4s, list) and len(ipv4s) > 1)
+        secondary_ipv6 = bool(isinstance(ipv6s, list) and len(ipv6s) > 1)
+        if secondary_ipv4 or secondary_ipv6:
+            if not config_secondary_ips:
+                LOG.debug(
+                    'Skipping secondary IP address setup.'
+                    ' Datasource configure_secondary_ips is False')
+            else:
+                ip_metadata = {
+                    'ipv4': {'cidr_key': 'subnet-ipv4-cidr-block',
+                             'has_secondary_ips': secondary_ipv4,
+                             'ips': ipv4s,
+                             'prefix_default': '24'},
+                    'ipv6': {'cidr_key': 'subnet-ipv6-cidr-block',
+                             'has_secondary_ips': secondary_ipv6,
+                             'ips': ipv6s,
+                             'prefix_default': '128'},
+                }
+                for ip_type, md_vals in ip_metadata.items():
+                    if not md_vals['has_secondary_ips']:
+                        continue
+                    cidr = nic_metadata.get(md_vals['cidr_key'])
+                    prefix = md_vals['prefix_default']
+                    if not cidr or len(cidr.split('/')) != 2:
+                        LOG.warning(
+                            'Could not parse %s %s. %s network'
+                            ' config prefix defaults to /%s',
+                            md_vals['cidr_key'], cidr, ip_type, prefix)
+                    else:
+                        prefix = cidr.split('/')[1]
+                    # We know we have > 1 ips for in metadata for this IP type
+                    for ip in md_vals['ips'][1:]:
+                        if 'addresses' not in dev_config:
+                            dev_config['addresses'] = []
+                        dev_config['addresses'].append(
+                            '{ip}/{prefix}'.format(ip=ip, prefix=prefix))
```

Comment:
What do you think about moving this to it's own function?

Reply:
good suggestion, added a get_secondary_addresses function and unit tests